m3u8/model.py

```diff
@@ -304,6 +304,10 @@ def dumps(self):
         if self.is_endlist:
             output.append('#EXT-X-ENDLIST')
 
+        # ensure that the last line is terminated correctly
+        if output[-1] and not output[-1].endswith('\n'):
+            output.append('')
```

Comment:
What is better here?

```python
output.append('')
```

or 

```python
output.append('\r\n')
```

Reply:
It needs to be `''` because of the `('\n').join(output)` that follows. Having a blank line causes the previous unterminated line and a new blank line to be joined using `\n`, giving the required result. Using `\r\n` causes that string to be joined, using `\n`, with the previous line, giving an additional unwanted `\n`.

An example from the test, printing `repr(manifest)`:
`\r\n`: `'#EXTM3U\n#EXT-X-TARGETDURATION:5220\n#EXTINF:5220,\nhttp://media.example.com/entire.ts\n#EXT-X-ENDLIST\n\r\n'` 👎 

`''`: `#EXTM3U\n#EXT-X-TARGETDURATION:5220\n#EXTINF:5220,\nhttp://media.example.com/entire.ts\n#EXT-X-ENDLIST\n` 👍 

I'll update the test to explicitly reject the unwanted result.

Another option I considered would have been to append a newline (eg `output[-1] += '\n'`), but I think the proposed way probably has a minor performance benefit.

Reply:
Oh, good catch! Thank you for the detailed answer.